src/api.c

```diff
@@ -211,14 +223,25 @@ gtk_layer_auto_exclusive_zone_is_enabled (GtkWindow *window)
 }
 
 void
-gtk_layer_set_keyboard_interactivity (GtkWindow *window, gboolean interacitvity)
+gtk_layer_set_keyboard_interactivity (GtkWindow *window, GtkLayerShellKeyboardInteractivity interactivity)
 {
     LayerSurface *layer_surface = gtk_window_get_layer_surface (window);
     if (!layer_surface) return; // Error message already shown in gtk_window_get_layer_surface
-    layer_surface_set_keyboard_interactivity (layer_surface, interacitvity);
+    uint32_t version = gtk_layer_get_layer_shell_version ();
+    if (interactivity >= GTK_LAYER_SHELL_KEYBOARD_ENTRY_NUMBER)
+    {
+        g_warning ("Invalid value for keyboard interactivity setting!\n");
```

Comment:
`g_return_if_fail(interactivity < GTK_LAYER_SHELL_KEYBOARD_ENTRY_NUMBER);` would be more compact here.